astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,251 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import re
+import warnings
+from collections import defaultdict, OrderedDict
+
+import numpy as np
+
+from . import Header, Card
+
+from ...coordinates import EarthLocation
+from ...table import Column
+from ...time import Time
```

Comment:
Agree with @saimn that we want to be cognizant of import issues (time and complexity).  In this case what we need to do is maintain a clear separation because the core `io.fits` (which we are trying to touch as little as possible) and the `Table` I/O layer which can be heavier in terms of imports.  

The only real problem I see now is the `from ..fitstime import is_time_column_keyword` in `io.fits.table.py`.  It would make sense to me to have `is_time_column_keyword` defined in `Table.py` and then within `fitstime.py` import from `.table`.  This maintains the existing layering where `table.py` is core and the connect / astropy Table methods are on the outside.